### PR TITLE
util,block,client,evm,vm: add EIP 7251 el triggered consolidations request type

### PR DESCRIPTION
**Pull request by g11tech**
add devnet 1 spec consolidations
for devnet 1 of prague (spec: https://notes.ethereum.org/@ethpandaops/pectra-devnet-1)

 ref
- https://eips.ethereum.org/EIPS/eip-7251

TODO: 
 - [x] fix failing spec
 - [x] modify newPayloadV4 spec

### CODE REVIEW

File: packages/client/src/rpc/validation.ts [hunk 1]
```diff
@@ -532,6 +532,54 @@ export const validators = {
...
+    return (requiredFields: string[] = ['sourceAddress', 'sourcePubkey', 'targetPubkey']) => {
+      return (params: any[], index: number) => {
+        if (typeof params[index] !== 'object') {
+          return {
+            code: INVALID_PARAMS,
+            message: `invalid argument ${index}: argument must be an object`,
+          }
+        }
+
+        const wt = params[index]
```
Comment (acolytec3) [inline packages/client/src/rpc/validation.ts:545]: What does `wt` indicate here?

File: packages/client/test/rpc/engine/newPayloadV4.spec.ts [hunk 2]
```diff
@@ -138,6 +143,11 @@ const electraGenesisContracts = {
     nonce: '1',
     code: '0x3373fffffffffffffffffffffffffffffffffffffffe1460575767ffffffffffffffff5f3511605357600143035f3511604b575f35612000014311604b57611fff5f3516545f5260205ff35b5f5f5260205ff35b5f5ffd5b5f35611fff60014303165500',
   },
+  '0x00b42dbF2194e931E80326D950320f7d9Dbeac02': {
+    balance: '0',
+    nonce: '1',
+    code: '0x3373fffffffffffffffffffffffffffffffffffffffe146098573615156028575f545f5260205ff35b36606014156101445760115f54600182026001905f5b5f82111560595781019083028483029004916001019190603e565b90939004341061014457600154600101600155600354806004026004013381556001015f35815560010160203581556001016040359055600101600355005b6003546002548082038060011160ac575060015b5f5b81811460f15780607402838201600402600401805490600101805490600101805490600101549260601b84529083601401528260340152906054015260010160ae565b9101809214610103579060025561010e565b90505f6002555f6003555b5f548061049d141561011d57505f5b6001546001828201116101325750505f610138565b01600190035b5f555f6001556074025ff35b5f5ffd',
```
Comment (acolytec3) [inline packages/client/test/rpc/engine/newPayloadV4.spec.ts:149]: Is this the code for this new system contract to perform consolidations? Can we provide some minimal commentary on how it works?

**Review by acolytec3 — CHANGES_REQUESTED** [review]
Looks fine though though a bit more comments in the couple of places I highlighted would be helpful

File: packages/client/src/rpc/validation.ts (hunk 1 above)
Comment (g11tech) [reply packages/client/src/rpc/validation.ts:545]: just a variable reference to the request

File: packages/client/test/rpc/engine/newPayloadV4.spec.ts (hunk 2 above)
Comment (g11tech) [reply packages/client/test/rpc/engine/newPayloadV4.spec.ts:149]: yes, its fully explained in EIP in detail

**Review by acolytec3 — APPROVED** [review]
LGTM

**Comment by holgerd77** [comment]
Ugh, we are getting new EIPs added to the code base really quickly 🤯.

Had a quick glance and seems everything ok, but we really shouldn't go quicker than this. 🙂 

HARD RULE: NO MORE THAN ONE EIP PER HOUR!!! 😂 😂 😂

**Pull request merged**